src/utils/staking.js

```diff
@@ -275,23 +277,39 @@ export class Staking {
     }
 
     async getValidators(accountIds) {
+        const { current_validators, next_validators, current_proposals } = await this.provider.validators()
+        const currentValidators = current_validators.map(({ account_id }) => account_id)
+        const nextValidators = current_proposals.map(({ account_id }) => account_id)
+        
         if (!accountIds) {
-            const { current_validators, next_validators, current_proposals } = await this.provider.validators()
-            const allValidators = [...current_validators, ...next_validators, ...current_proposals]
-            accountIds = [...new Set(allValidators.map(({ account_id }) => account_id))]
+            const rpcValidators = [...current_validators, ...next_validators, ...current_proposals].map(({ account_id }) => account_id)
+
+            // TODO use indexer - getting all historic validators from raw GH script .json
+            const networkId = this.provider.connection.url.indexOf('mainnet') > -1 ? 'mainnet' : 'testnet'
+            if (!ghValidators) {
+                ghValidators = (await fetch(`https://raw.githubusercontent.com/frol/near-validators-scoreboard/scoreboard-${networkId}/validators_scoreboard.json`).then((r) => r.json()))
+                .map(({ account_id }) => account_id)
+            }
+            
+            accountIds = [...new Set([...rpcValidators, ...ghValidators])]
+                .filter((v) => v.indexOf('nfvalidator') === -1 && v.indexOf(networkId === 'mainnet' ? '.near' : '.m0') > -1)
```

Comment:
@mattlockyer why do we have to filter out here? Isn't scoreboard already filtered by network ID?

Reply:
No. It includes a lot of legacy validators that we don't want to process. It would just mean another check against their account and additional RPC call.